File: time2sex.py
#!/usr/bin/python

import sys

def time_to_seconds(time):
    try:
        time = [ int(i) for i in time.split(":") ]
    except ValueError:
        print "Please provide valid input"
        return "No"
    # python golf
    return sum( [ [1,60,3600][i]*j for i, j in enumerate(reversed(time[:3]))] )
   
if __name__ == "__main__":
    if len(sys.argv) == 2:
        print "=== %s seconds ===" % time_to_seconds(sys.argv[1])
    else:
        print "Please enter a command line arg"
        print "HH:MM:SS"
        print "EX:"
        print "    time2sex 1:23:45"

File: time2sex.rb


def time2sex(time)
    secs = [1, 60, 3600]
    time = time.split(":").reverse
    seconds = 0

    for i in 0..time.length-1
        seconds +=  secs[i] * time[i].to_i
    end

    return "=== " + seconds.to_s + " seconds ==="
end

if ARGV.length == 1
    puts time2sex(ARGV[0])
else
    puts "Please enter a command line arg"
    puts "HH:MM:SS"
    puts "EX:"
    puts "    time2sex 12:34:56"
end
